spyderlib/widgets/file_switcher.py

```diff
@@ -0,0 +1,405 @@
+# -*- coding: utf-8 -*-
+#
+# Augmented file switcher by DM, 2015 (based on original in widgets/editor.py)
+# Licensed under the terms of the MIT License
+# (see spyderlib/__init__.py for details)
+
+from __future__ import print_function
+
+from spyderlib.baseconfig import _
+from spyderlib.qt.QtGui import (QVBoxLayout, QHBoxLayout,
+                                QListWidget, QListWidgetItem,
+                                QDialog, QLineEdit, QKeyEvent,
+                                QStyledItemDelegate, QStyleOptionViewItemV4, 
+                                QApplication, QTextDocument, QStyle,
+                                QAbstractTextDocumentLayout, QToolButton,
+                                QToolTip)
+from spyderlib.qt.QtCore import (Signal, Qt,  QObject,
+                                  QSize, QEvent, QPoint)
+
+import os
+import os.path as osp
+
```

Comment:
It would be better if the import follow the PEP8 conventions..., first standard library, then external libraries then local imports like : 

```python

from __future__ import print_function
import os
import os.path as osp

from spyderlib.qt.QtCore import Signal, Qt,  QObject QSize, QEvent, QPoint
from spyderlib.qt.QtGui import (QVBoxLayout, QHBoxLayout,
                                QListWidget, QListWidgetItem,
                                QDialog, QLineEdit, QKeyEvent,
                                QStyledItemDelegate, QStyleOptionViewItemV4, 
                                QApplication, QTextDocument, QStyle,
                                QAbstractTextDocumentLayout, QToolButton,
                                QToolTip)

# Local imports
from spyderlib.baseconfig import _
from spyderlib.guiconfig import  new_shortcut
from spyderlib.py3compat import to_text_string
from spyderlib.utils.qthelpers import get_std_icon

```